crates/miden-lib/asm/account_components/multisig_rpo_falcon_512.masm

```diff
@@ -16,26 +16,31 @@ const.UNAUTHORIZED_EVENT=event("miden::auth::unauthorized")
 
 # Storage Layout Constants
 #
-# ┌─────────────────────────────┬──────────┬──────────────┐
-# │    THRESHOLD & APPROVERS    │ PUB KEYS │ EXECUTED TXS │
-# │           (slot)            │   (map)  │    (map)     │
-# ├─────────────────────────────┼──────────┼──────────────┤
-# │              0              │    1     │      2       │
-# └─────────────────────────────┴──────────┴──────────────┘
-
-# The slot in this component's storage layout where both the signature threshold
-# and number of approvers are stored as [threshold, num_approvers, 0, 0].
+# ┌───────────────────────────────┬──────────┬──────────────┬──────────────┐
+# │ THRESHOLD & APPROVERS CONFIG  │ PUB KEYS │ EXECUTED TXS │  PROC ROOTS  │
+# │           (slot)              │   (map)  │    (map)     │    (map)     │
+# ├───────────────────────────────┼──────────┼──────────────┼──────────────┤
+# │              0                │    1     │      2       │      3       │
+# └───────────────────────────────┴──────────┴──────────────┴──────────────┘
+
+# The slot in this component's storage layout where the signature threshold,
+# number of approvers are stored as
+# [threshold, num_approvers, 0, 0].
 # The threshold is guaranteed to be less than or equal to num_approvers.
 const.THRESHOLD_CONFIG_SLOT=0
 
 # The slot in this component's storage layout where the public keys map is stored.
-# Map entries: [key_index, 0, 0, 0] => owner_public_key
+# Map entries: [key_index, 0, 0, 0] => APPROVER_PUBLIC_KEY
 const.PUBLIC_KEYS_MAP_SLOT=1
 
 # The slot in this component's storage layout where executed transactions are stored.
 # Map entries: transaction_message => [is_executed, 0, 0, 0]
 const.EXECUTED_TXS_SLOT=2
 
+# The slot in this component's storage layout where procedure thresholds are stored.
+# Map entries: PROC_ROOT => [threshold, 0, 0, 0]
+const.PROC_THRESHOLD_ROOTS_SLOT=3
```

Comment:
nit (feel free to ignore): I would maybe make this storage slot 2 and change executed txs to be in storage slot 3.